### PR TITLE
add an option to support pkce

### PR DESCRIPTION
**Pull request by huayang-codaio**
PTAL @spencer-codaio @jonathan-codaio 

https://staging.coda.io/d/Packs-Platform-IA-go-packs_d36WK4zgrYx/Weekly-status-updates_suaOK?utm_source=slack&utm_content=comment_notification#All-team-status-updates_tuDwY/r63

### CODE REVIEW

**Review by spencer-codaio — APPROVED** [review]
nice thank you!

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 This changelog keeps track of all changes to the Packs SDK. We follow conventions from [keepachangelog](https://keepachangelog.com/en/1.0.0/).
 
+## [Unreleased]
+
+### Added
+
+- Added "useProofKeyForCodeExchange" option to OAuth2 authentication to support PKCE extention. While it's optionally supported by most OAuth2 providers, it might be required by some websites (e.g. Twitter).
```
Comment (spencer-codaio) [inline CHANGELOG.md:9]: hmm wait im not sure how we got a 0.11.0 header even tho it's not released yet but do you want to move it into 0.11.0?

Comment (spencer-codaio) [reply CHANGELOG.md:9]: oh this is from jonathan's last change. Wonder if we should include this in there too but not big deal

Comment (huayang-codaio) [reply CHANGELOG.md:9]: I don't want to further block 0.11.0 release. This might go into 0.12.0.

Comment (jonathan-codaio) [inline CHANGELOG.md:9]: Nit, `extension`

File: types.ts [hunk 2]
```diff
@@ -419,6 +419,18 @@ export interface OAuth2Authentication extends BaseAuthentication {
    * that should contain the token.
    */
   tokenQueryParam?: string;
+
+  /**
+   * Option to apply PKCE (Proof Key for Code Exchange) OAuth2 extension. With PKCE extention,
```
Comment (jonathan-codaio) [inline types.ts:424]: Ditto.

**Review by jonathan-codaio — APPROVED** [review]

**Comment by huayang-codaio** [comment]
actual changes: https://github.com/coda/coda/pull/71298

**Pull request merged**